### DISCUSSION TITLE
Alignment in Cursive

### DISCUSSION THREAD

**Discussion by kajuburfi** [gyscos/cursive#626] [Q&A] [answered]
I wanted to make an application in which a command mode (like vim or kakoune's) is present..
But, I am not able to align the dialog box(of an <kbd>EditView</kbd>) at the bottom of the terminal screen.

Is it possible to do so?
If so, how? 

Thanks in advance

**Comment by gyscos** [comment, marked as answer]
Hi, and thanks for the question!

Right now there are two ways to do it:
* The manual way: use `StackView::add_layer_at` to manually put the new layer at the bottom of the screen, and use `StackView::reposition_layer` to keep it at the bottom when you detect a resize event.
* The automatic way: write a custom view wrapper that requests the full width and height, but keep its child at the bottom.

https://docs.rs/cursive-aligned-view/latest/cursive_aligned_view/ may be an implementation of the second solution.

**Reply by kajuburfi** [reply]
Thanks a lot! Will try so!

**Reply by kajuburfi** [reply]
Can you give an example for the same?
I am getting confused..

Thanks in advance.

**Reply by gyscos** [reply]
There's an example in the doc for that crate:

```rust
use cursive::{self, CursiveExt};
use cursive::view::Resizeable;
use cursive::views::{Panel, DummyView};

use cursive_aligned_view::AlignedView;

fn main() {
    let mut siv = cursive::default();

    let panel = Panel::new(DummyView)
        .title("Hello, world!")
        .full_width()
        .align_bottom_center();

    siv.add_fullscreen_layer(panel);
    // siv.run()
}
```
